### ISSUE TITLE
New Mexico income tax integration test failure

### ISSUE THREAD

**Issue by martinholmer** [PolicyEngine/policyengine-us#2729]
@PavelMakarchuk,  When you asked me to test the newly developed NM income tax module, you said you wanted to be involved in the process.  So, here is your chance.

The newest results in issue #993 show that there are many NM differences in the simplest `p21` sample between PolicyEngine-US 0.396.0 and the latest patched version of TAXSIM35.  Below is a test case from that `p21` sample.  So, now we are at step 6 in the process described in topic #2389: the two models produce different results, but it isn't clear which model is wrong.

Use the methods described in step 6 to answer that question.

Here are the results from running the test:
```
  Test 'Tax unit with taxsimid 86990 in p21.its.csv and p21.ots.csv':
    nm_income_tax@2021: [-250.] differs from -941.010009765625
                        with an absolute margin [691.01] > 0.01
```
And here is the YAML-formatted test:
```
- name: Tax unit with taxsimid 86990 in p21.its.csv and p21.ots.csv
  absolute_error_margin: 0.01
  period: 2021
  input:
    people:
      person1:
        is_tax_unit_head: true
        age: 66
        employment_income: 1010
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
      person2:
        is_tax_unit_spouse: true
        age: 66
        employment_income: 8010
        ssi: 0  # not in TAXSIM35
        state_supplement: 0  # not in TAXSIM35
        wic: 0  # not in TAXSIM35
    tax_units:
      tax_unit:
        members: [person1, person2]
        premium_tax_credit: 0  # not in TAXSIM35
        local_income_tax: 0  # not in TAXSIM35
        state_sales_tax: 0  # not in TAXSIM35
        nm_2021_income_rebate: 0  # not in TAXSIM35
        nm_additional_2021_income_rebate: 0  # not in TAXSIM35
        nm_supplemental_2021_income_rebate: 0  # not in TAXSIM35
    spm_units:
      spm_unit:
        members: [person1, person2]
        snap: 0  # not in TAXSIM35
        tanf: 0  # not in TAXSIM35
    households:
      household:
        members: [person1, person2]
        state_code: NM
  output:  # expected results from patched TAXSIM35 2023-07-27 version
    nm_income_tax_before_refundable_credits: 0.00
    nm_income_tax: -941.01
```

**Comment by MaxGhenis** [comment]
Thanks @martinholmer - though in your output you use `mn_income_tax_before_refundable_credits` instead of `nm_income_tax_before_refundable_credits`

**Comment by martinholmer** [comment]
Fixed typo in YAML test case.  Test results are unchanged.

**Comment by MaxGhenis** [comment]
Thanks - in taxsim.app it shows $975 instead of $941 ([link](https://taxsim.app/?year=2021&mstat=married&debug=1&settings.numeric=1&state=32&page=66&sage=66&pwages=1010&swages=8010)), but should give us enough to start debugging.
<img width="433" alt="image" src="https://github.com/PolicyEngine/policyengine-us/assets/6076111/f64ba032-a787-4a07-8fed-5cb41a7b9846">

**Comment by MaxGhenis** [comment]
I just looked through the taxsim fortran and didn't see `socredit`. Maybe this was renamed?

**Comment by martinholmer** [comment]
@MaxGhenis asked:
> can you tell from the taxsim code what socredit means?  
The `taxsim.app` lists it as a source of $699 for this filer, but I don't see it in https://taxsim.nber.org/taxsim35.

Ever since I first looked at `taxsim.app` and saw how wrong it was (no spouse earnings, and hence, incorrect CDCC amounts, for example), I have ignored it.    Perhaps `socredit` is something `taxsim.app` has made up.  Why don't you ask the `taxsim.app` author?

**Comment by MaxGhenis** [comment]
taxsim.app now has spouse earnings. It simplifies some logic for the web interface but doesn't introduce any logic beyond taxsim.
<img width="578" alt="image" src="https://github.com/PolicyEngine/policyengine-us/assets/6076111/699d9824-0dd4-4aa2-bd31-8a74c0b49dce">

Would it be inconvenient to export more results from taxsim in your script?

**Comment by MaxGhenis** [comment]
I ran this scenario through the [nber taxsim35 web app](https://taxsim.nber.org/taxsim35/redirect.cgi), and got the below results. The -$975 matches taxsim.app (@martinholmer do you know why your patched version would give -$941?) and it doesn't provide further detail around the non-EITC credits.
```
State Tax Calculation:      
     30. Household Income             9020.00     9021.00
     31. Imputed Rent                    0.00        0.00
     32. AGI                          9020.00     9021.00
     33. Exemptions                   5000.00     5000.00
     34. Standard Deduction          27800.00    27800.00
     35. Itemized Deductions             0.00        0.00
     36. Taxable Income                  0.00        0.00
         Tax before credits              0.00        0.00
     37. Property Tax Credit             0.00        0.00
     38. Child Care Credit               0.00        0.00
     39. EIC                           276.01      276.04
         Energy|Fuel Credit              0.00        0.00
         Child Tax Credit                0.00        0.00
     40. Total Credits                 975.35      975.34
     41. Bracket Rate                    1.70        1.70
         State Tax after Credits      -975.35     -975.34
```

**Comment by feenberg** [comment]
How many exemptions are you using in the calculation of the low income 
allowance?

https://taxsim.nber.org/historical_state_tax_forms/NM/2021/2021ycred.pdf

It looks like 2e and 2f on:

https://taxsim.nber.org/historical_state_tax_forms/NM/2021/2021pit-rc.pdf

give 2 extra exemptions for each 65+ taxpayer. Is that right? Seems odd, 
but taxsim does that.

Dan

**Comment by MaxGhenis** [comment]
Thank you Dan, yes we missed that. Will work on it in https://github.com/PolicyEngine/policyengine-us/issues/2735 and see how it affects the result.

**Comment by feenberg** [comment]
The way to find these things is to buy a copy of "Taxact All-States 
Self-Employed" for about $100 and enter let it show you the calculation.

I'd love to have your test data. Any chance you would share it with your 
numbers attached for comparisons?

I am updating to 2022 now, but some of the updates includes fixes to 2021 
and earlier, and it would be useful for me to have your tests too.

Dan

**Comment by MaxGhenis** [comment]
Thanks Dan, we'll pick up Taxact. Martin may be able to provide the test data.

In https://github.com/PolicyEngine/policyengine-us/pull/2736 we added an integration test for this case, which has $9,020 income and (now) 6 exemptions. But following this table from the [2021 tax form](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/608063af-0c4c-4f28-ba42-cf40e04557ca/PITbook2021.pdf#page=60), we assign a $665 credit. The modified gross income concept doesn't have any subtractions that would bring it down to below $9,000 resulting in a $700 credit. Since there's no property tax, rent, or medical expenses we don't see other credits that could create the $35 discrepancy. Any ideas Dan?

<img width="417" alt="image" src="https://github.com/PolicyEngine/policyengine-us/assets/6076111/1572293b-46f0-4571-880e-9febc32df4d8">

**Comment by martinholmer** [comment]
Issue #2729 has been resolved by two merged pull requests: #2732 and #2736.
